### DISCUSSION TITLE
Feature extraction and projections

### DISCUSSION THREAD

**Discussion by msoiron** [stella-cv/stella_vslam#411] [Q&A] [answered]
Hi,
as I understand the orb feature extractor is applied to each frame without any preprocessing.

As we are using equirectangular frames, is it possible that the distortion in the top and bottom part of the video would lead to problems during feature detection in those parts and is that maybe the reason why they are masked in the provided yaml settings?

Would it be possible to unproject the whole frame and do the feature extraction within the spherical coordinate space (hence without distortions) and if yes, would that increase performance?

**Comment by ymd-stella** [comment, marked as answer]
> As we are using equirectangular frames, is it possible that the distortion in the top and bottom part of the video would lead to problems during feature detection in those parts and is that maybe the reason why they are masked in the provided yaml settings?

Yes.

> Would it be possible to unproject the whole frame and do the feature extraction within the spherical coordinate space (hence without distortions)

[If needed, users can implement extra camera models (e.g. dual fisheye, catadioptric) with ease.](https://github.com/stella-cv/stella_vslam/blob/main/README.md#features)

> and if yes, would that increase performance?

I don't think the performance will be much different. If you are unsatisfied with the performance, there are other things you should be doing more of.

**Reply by msoiron** [reply]
Thank you for the quick answer.

For the camera model we could only change the rectangular mask. We already experimented which removing the upper part, but that didn't make much difference. And if its there for a reason that might be a bad idea?

>If you are unsatisfied with the performance, there are other things you should be doing more of.

Could you throw us some hinds? We already tried to optimise resolution and the rate key frames are collected.

**Reply by ymd-stella** [reply]
What do you measure performance by?

**Reply by msoiron** [reply]
Most of our videos are from construction sights. We get the absolute positions from the beginning and the end points of the videos and we have a floor plan.

We align the SLAM trajectory with that floor plan. From that we can see by eye if the trajectory goes through the rooms shown in the video. Which is not always the case. Sometimes the trajectory difts by several meters and we sometimes get scale drift.

Of course we cannot expect every video to be perfect, but we are looking for possibilities to improve our current results.

**Reply by ymd-stella** [reply]
If the scene is an outdoor environment scene with an open field of view, you might try reducing baseline_dist_thr_ratio. Increasing the number of keyframes will not increase landmarks if there are not enough baselines. This parameter adjusts that threshold.

**Reply by ymd-stella** [reply]
Adjust num_covisibilities_for_landmark_generation and num_covisibilities_for_landmark_fusion. Long baseline length matching improves accuracy.

**Reply by msoiron** [reply]
Okay thank you. We try that.

**Reply by chongyaaa** [reply]
Reducing baseline_dist_thr_ratio, will reduce the baseline?

**Reply by ymd-stella** [reply]
See https://github.com/stella-cv/stella_vslam/blob/f94dbda3e6237f4a77f413bcf33282f4e91ac1c4/src/stella_vslam/mapping_module.cc#L278

**Reply by chongyaaa** [reply]
if (baseline_dist < baseline_dist_thr_ratio_ * median_depth_in_ngh) {
                continue;
            }
Reduce baseline_dist_thr_ratio_,will increase landmarks.

**Reply by chongyaaa** [reply]
> 
Hello, how about the effect of your experiment?

**Reply by msoiron** [reply]
Hi,
I tried all those parameters but I could not see that it makes a large difference. I think the drift we observed is most likely due to poor lighting in one region of our video. So there is most likely not much we can do about it.
